crates/next-core/src/middleware.rs

```diff
@@ -34,11 +35,66 @@ pub async fn get_middleware_module(
     let is_proxy = userland_path.file_stem() == Some("proxy");
     let page_path = if is_proxy { "/proxy" } else { "/middleware" };
 
+    // Validate that the module has the required exports
+    if let Some(ecma_module) =
+        Vc::try_resolve_sidecast::<Box<dyn EcmascriptChunkPlaceable>>(*userland_module).await?
+    {
+        let exports = ecma_module.get_exports().await?;
+
+        // Check if the module has the required exports
+        let has_valid_export = match &*exports {
+            // ESM modules - check for named or default export
+            EcmascriptExports::EsmExports(esm_exports) => {
+                let esm_exports = esm_exports.await?;
+                let has_default = esm_exports.exports.contains_key(&rcstr!("default"));
+                let expected_named = if is_proxy {
+                    rcstr!("proxy")
+                } else {
+                    rcstr!("middleware")
+                };
```

Comment:
I think this should work?

```suggestion
                let has_default = esm_exports.exports.contains_key("default");
                let expected_named = if is_proxy {
                    "proxy"
                } else {
                    "middleware"
                };
```

- `RcStr` implements `Borrow<str>`: https://turbopack-rust-docs.vercel.sh/rustdoc/turbo_rcstr/struct.RcStr.html#impl-Borrow%3Cstr%3E-for-RcStr
- `contains_key(&self, k: &Q)` requires `K: Borrow<Q>`: https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.contains_key

So it should be possible for `Q` to be `str` in this case.